Gems/LmbrCentral/Code/Source/Shape/BoxShape.cpp

```diff
@@ -267,25 +267,61 @@ namespace LmbrCentral
         }
 
         // transform to world space
-        return m_currentTransform.TransformPoint(AZ::Vector3(x, y, z));
+        AZ::Transform worldTransformWithoutScale = m_currentTransform;
+        const float entityScale = worldTransformWithoutScale.ExtractUniformScale();
+        const AZ::Vector3 scaledTranslationOffset = entityScale * m_currentNonUniformScale * m_boxShapeConfig.m_translationOffset;
+
+        return worldTransformWithoutScale.TransformPoint(AZ::Vector3(x, y, z) + scaledTranslationOffset);
+    }
+
+    AZ::Vector3 BoxShape::GetTranslationOffset()
+    {
+        return m_boxShapeConfig.m_translationOffset;
+    }
+
+    void BoxShape::SetTranslationOffset(const AZ::Vector3& translationOffset)
+    {
+        if (!IsShapeComponentTranslationEnabled())
+        {
+            return;
+        }
+
+        bool shapeChanged = false;
+        {
+            AZStd::unique_lock lock(m_mutex);
+            if (!m_boxShapeConfig.m_translationOffset.IsClose(translationOffset))
+            {
+                m_boxShapeConfig.m_translationOffset = translationOffset;
+                m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::ShapeChange);
+                shapeChanged = true;
+            }
+        }
+
+        if (shapeChanged)
+        {
+            ShapeComponentNotificationsBus::Event(
+                m_entityId, &ShapeComponentNotificationsBus::Events::OnShapeChanged,
+                ShapeComponentNotifications::ShapeChangeReasons::ShapeChanged);
+        }
     }
 
     void BoxShape::BoxIntersectionDataCache::UpdateIntersectionParamsImpl(
         const AZ::Transform& currentTransform, const BoxShapeConfig& configuration, const AZ::Vector3& currentNonUniformScale)
     {
         AZ::Transform worldFromLocalNormalized = currentTransform;
```

Comment:
Why does this transform get copied instead of using currentTransform directly? Unless I'm missing something it doesn't seem to be changing any of the values.

Reply:
I think ExtractUniformScale on the next line will modify it.

Reply:
Eeek, yep, you're right, I did NOT read "Extract" as a destructive action. That's a method name that could use a bit more verbosity. :(

Reply:
Because of the call to `worldFromLocalNormalized.ExtractUniformScale()` below which mutates `worldFromLocalNormalized`